### ISSUE TITLE
Number of AvgNumGroundtruthBoxesPerImage is always 100

### ISSUE THREAD

**Issue by christian-lanius** [tensorflow/models#7469]
### System information
- **What is the top-level directory of the model you are using**: object_detection/models/research/object_detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Kubuntu 18.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.14
- **Bazel version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: Cuda 10.0 cuDnn7
- **GPU model and memory**: RTX 2080TI
- **Exact command to reproduce**: 
```
PIPELINE_CONFIG_PATH=/home/lanius/code/object_detection/trained_models/ssd_resnet50_v1_fpn_shared_box_predictor/pipeline_clean_few_classes.config
MODEL_DIR=/home/lanius/code/object_detection/results/ssd_resnet50_v1_fpn/clean_few_classes
SAMPLE_1_OF_N_EVAL_EXAMPLES=1
python object_detection/model_main.py \
    --pipeline_config_path=${PIPELINE_CONFIG_PATH} \
    --model_dir=${MODEL_DIR} \
    --sample_1_of_n_eval_examples=$SAMPLE_1_OF_N_EVAL_EXAMPLES \
    --alsologtostderr 
```

### Describe the problem
I am attempting to train the ssd_resnet_50_fpn_coco object detector on a simplified coco dataset. However, the loss metrics reported by tensorboard look off:
![image](https://user-images.githubusercontent.com/1444164/63274223-b8f28180-c29f-11e9-885a-43810123ad0e.png)
The number of AvgNumGroundtruthBoxesPerImage is always 100, the maximum number of output boxes from the model. I use the following label map:
```
item {
  name: "/m/01g317"
  id: 1
  display_name: "person"
}
item {
  name: "/m/0199g"
  id: 2
  display_name: "bicycle"
}
item {
  name: "/m/0k4j"
  id: 3
  display_name: "car"
}
item {
  name: "/m/04_sv"
  id: 4
  display_name: "motorcycle"
}
item {
  name: "/m/01bjv"
  id: 5
  display_name: "bus"
}
item {
  name: "/m/07r04"
  id: 6
  display_name: "truck"
}
item {
  name: "/m/015qff"
  id: 7
  display_name: "traffic light"
}
item {
  name: "/m/02pv19"
  id: 8
  display_name: "stop sign"
}

```
Indeed in the tensorboard display, there are some images with no GT bboxes:
![image](https://user-images.githubusercontent.com/1444164/63274513-328a6f80-c2a0-11e9-8b4d-de0c6af34dcd.png)
While the loss in general decreases, the performance of the detector is very poor. After 15000 steps with a batchsize of 8:
```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.015
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.031
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.015
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.004
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.031
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.018
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.025
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.026
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.006
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.059
```

**Comment by zhangqianYY** [comment]
i meet same priblem ,do you  have solve it?

**Comment by christian-lanius** [comment]
Hi Zhangqian,
After tuning some hyperparameters, I got better performance (not super well, but enough for my needs). I think the problem is mainly due a reporting problem, and not an actual performance problem.
I think at this point, with YOLOv3 impressive performance, there is little reason to use a SSD architecture. YMMV

**Comment by jveitchmichaelis** [comment]
This is due to a parameter in the pipeline.config called `unpad_groundtruth_tensors` which I assume pads this list to whatever the maximum number of boxes is. 

If you set this from `false` to `true`, you'll get the right number. Though bear in mind this shouldn't have any effect on training. It's simply tracking how many ground truth boxes on average you have in your dataset.
